packages/driver/src/cy/commands/screenshot.ts

```diff
@@ -338,8 +332,10 @@ const takeScreenshot = (Cypress, state, screenshotConfig, options: TakeScreensho
 
       return null
     })
-    .then(() => {
-      return sendAsync('before:screenshot', getOptions(true))
+    .then(async () => {
+      // we need to await this so that JS running to prepare for the screenshot
+      // is fully completed before we start measuring the viewport etc
+      await send('before:screenshot', getOptions(true), () => {})
```

Comment:
Does this actually make a difference? Either way, `before` will end up returning a `Promise`, which Cypress should already be waiting for by default. I'm not sure this changes behavior (it doesn't look like it should, at least). I could be missing something.